### ISSUE TITLE
Regression for editing links in an Email: can't change link properties anymore

### ISSUE THREAD

**Issue by LordRembo** [mautic/mautic#13362]
### Mautic Version

5.0.x series

### Way of installing

I cloned the Mautic repo (not recommended or supported)

### PHP version

8.0

### What browsers are you seeing the problem on?

Not relevant

### What happened?

There is a regression in functionality, for the text editing popup dialog, between M5 and M4. Probably related to updating from cke4 to cke5.
There are no advanced settings for editing links in M5 (can only add or remove the href), while you can set things like a target, title and various other properties in M4. Specifically, setting 'target' and 'title' are useful to have.

### How can we reproduce this issue?

Step 1: Make a new email in Mautic 5
Step 2: Add a Text block to your mail (I used an MJML template)
Step 3: Add a link to that text
Step 4: Select that text and click the chain-icon, you only get the option to change the href.
Step 5: Compare to a similar mail and text block in Mautic 4, where adding or editing a link gives you a lot more options

![link-cke5](https://github.com/mautic/mautic/assets/758957/753d86b8-bc9d-49dd-ad0f-8a0dc42064f3)

![link-cke4-a](https://github.com/mautic/mautic/assets/758957/2296e20e-4899-465b-97ae-168f64a29a62)
![link-cke4b](https://github.com/mautic/mautic/assets/758957/3e071731-b47f-4b44-9807-91885f662d43)
![link-cke4c](https://github.com/mautic/mautic/assets/758957/b848830c-72e3-4d99-b0b9-3063493dc561)

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by jvir** [comment]
I think other advanced properties are important as well, for example Style to be able to remove the decoration from the links.

**Comment by LordRembo** [comment]
@jvir I agree there could be more properties. Probably best if we can compile and review a list. Adding stuff to CKE5 is a nightmare so depending on what thing needs to be added, it could be easy (like that 'target', was a easy option to enable) or difficult (eg. a prop that requires an input field will probably need to be done via a new plugin). 

So far, we have these that seem useful. Anything else you can thing of that should be added?

- setting target on a link (which has a PR already)
- setting the title attribute
- adding inline styling on a link

One way around it, which is less user-friendly, could be to enable 'source code' in that CKE popup. Maybe that could be done as a temporary workaround?

**Comment by LordRembo** [comment]
To give an update on this ticket, it's still open but 2 things have been merged since then:

1. you can now set a link target
2. the Editor doesn't filter out spans and link properties anymore

What is still needed:

- being able to set a title attribute on a link
- being able to add inline styling to a link -> discussed the option to simply enable the 'source code' button but people rightly pointed out that's adding a lot of risk to break stuff, so not really a viable solution

**Comment by RCheesley** [comment]
Thanks for the update on the issue @LordRembo - much appreciated!

**Comment by robbiewu003** [comment]
I cannot change my link url.
1. select the text with a link or no a link
2. click the link button in edior.
3. no popup window.

chrome  131.0.6778.109
mautic 5.2
It works on mautic 5.0

**Comment by LordRembo** [comment]
@robbiewu003 That sounds like a new bug, that's not really related to this ticket. Can you make a new one please?  
To make debugging easier, could you add a bit of extra info:

- what theme you are using (so we know if it's an MJML or an HTML based one)
- screenshots of the editor as you are selecting or editing the text

**Comment by LordRembo** [comment]
This should be reopened and addressed, it's relevant for all Mautic versions.
